geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/Services.java

```diff
@@ -206,6 +209,18 @@ public void start() throws MemberStartupException {
       this.joinLeave.started();
       this.healthMon.started();
       this.manager.started();
+
+      if (membershipLocator != null) {
+        /*
+         * Now that all the services have started we can let the membership locator know
+         * about them. We must do this before telling the manager to joinDistributedSystem()
+         * later in this method
+         */
+        final MembershipLocatorImpl locatorImpl =
+            (MembershipLocatorImpl) this.membershipLocator;
+        locatorImpl.setServices(this);
```

Comment:
This logic here replaces what used to be implemented by callers in their `LifecycleListener.started()` (overridden) methods. Nice thing about this is it's implemented in one place so clients of the membership subsystem can't screw it up